module/documents/activity/attack.mjs

```diff
@@ -277,9 +250,42 @@ export default class AttackActivity extends ActivityMixin(AttackActivityData) {
      * @param {object} data
      * @param {AttackActivity} data.subject  The activity that performed the attack.
      */
-    Hooks.callAll("dnd5e.postRollAttack", [roll], { subject: this });
+    Hooks.callAll("dnd5e.postRollAttack", rolls, { subject: this });
+
+    return rolls;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Configure a roll config for each roll performed as part of the attack process. Will be called once per roll
+   * in the process each time an option is changed in the roll configuration interface.
+   * @param {Partial<D20RollConfiguration>} [initialRoll]  Initial roll passed to the rolling method.
+   * @param {D20RollProcessConfiguration} process          Configuration for the entire rolling process.
+   * @param {D20RollConfiguration} config                  Configuration for a specific roll.
+   * @param {FormDataExtended} [formData]                  Any data entered into the rolling prompt.
+   * @param {number} index                                 Index of the roll within all rolls being prepared.
+   */
+  _buildAttackConfig(initialRoll, process, config, formData, index) {
+    if ( formData?.has("ammunition") ) process.ammunition = formData.get("ammunition");
+    if ( formData?.has("attackMode") ) process.attackMode = formData.get("attackMode");
+    if ( formData?.has("mastery") ) process.mastery = formData.get("mastery");
```

Comment:
Why do we need to update `process` here? Shouldn't it be immutable?

Reply:
Hmm, right now `getAttackData` fetches information on the ammunition from the attack process config (and potentially attack mode in a future version). I suppose we could pass those options directly rather than passing in the whole config object.